File: RB101_programming_foundations/lesson_5/5_practice_problems/ex_08.rb
# Using the each method, write some code to output all of the vowels from the strings.

hsh = {first: ['the', 'quick'], second: ['brown', 'fox'], third: ['jumped'], fourth: ['over', 'the', 'lazy', 'dog']}

hsh.each do |_, array|
  array.join('').each_char do |char|
    if 'aeiouAEIOU'.include?(char)
      puts char
    end
  end
end

File: JS101_programming_foundations/lesson_5/practice_problems/ex_08.js
// Using the forEach method, write some code to output all vowels from the strings in the arrays.Don't use a for or while loop.

let obj = {
  first: ['the', 'quick'],
  second: ['brown', 'fox'],
  third: ['jumped'],
  fourth: ['over', 'the', 'lazy', 'dog'],
};

Object.values(obj).forEach(arr => {
  let matches = arr.join(' ').match(/[aeiou]/gi) || false;
  if (matches) matches.forEach(char => console.log(char));
});
